tmt/steps/execute/__init__.py

```diff
@@ -331,7 +331,38 @@ def check_result_file(self, test: "tmt.Test") -> "tmt.Result":
         except KeyError:
             data['result'] = "error"
             data['note'] = f"invalid test result '{result}' in result file"
-        return tmt.Result(data, test=test)
+        return [tmt.Result(data, test=test)]
+
+    def check_custom_results(self, test: "tmt.Test") -> List["tmt.Result"]:
+        """
+        Process custom results.yaml file created by the test itself.
+        """
+        self.debug("Processing custom results.yaml file created by the test itself.")
+
+        custom_results_path = os.path.join(
+            self.data_path(test, full=True),
+            tmt.steps.execute.TEST_DATA,
+            'results.yaml')
+
+        if not os.path.exists(custom_results_path):
+            # Missing results.yaml means error result, but tmt contines with other tests
+            return [tmt.Result(
+                data={
+                    'note': f"Custom results file '{custom_results_path}' not found.",
```

Comment:
```suggestion
                    'note': f"custom results file '{custom_results_path}' not found",
```

All result notes so far are lower case, so let's make it consistent.